MedraPlugin.php

```diff
@@ -0,0 +1,394 @@
+     * ]
+     */
+    public function addAllowedObjectTypes(string $hookName, array $args): bool
+    {
+        $objectTypeOptions = &$args[0];
+        $allowedTypes = $this->getAllowedDoiTypes();
+
+        $objectTypeOptions = array_map(function ($option) use ($allowedTypes) {
+            if (in_array($option['value'], $allowedTypes)) {
+                $option['allowedBy'][] = $this->getName();
+            }
+            return $option;
+        }, $objectTypeOptions);
+
+        return Hook::CONTINUE;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAllowedDoiTypes(): array
+    {
+        return [
+            Repo::doi()::TYPE_PUBLICATION,
+            Repo::doi()::TYPE_ISSUE,
+        ];
+    }
+
+    /**
+     * Adds mEDRA specific info to Repo::doi()->markRegistered()
+     *
+     * @param string $hookName Doi::markRegistered
+     *
+     */
+    public function editMarkRegisteredParams(string $hookName, array $args): bool
+    {
+        $editParams = &$args[0];
+        $editParams[$this->getFailedMsgSettingName()] = null;
+        return HOOK::CONTINUE;
+    }
+
+    /**
+     * Get key for retrieving error message if one exists on DOI object
+     */
+    public function getErrorMessageKey(): ?string
+    {
+        return $this->getFailedMsgSettingName();
+    }
+
+    /**
+     * Get key for retrieving registered message if one exists on DOI object
+     *
+     */
+    public function getRegisteredMessageKey(): ?string
+    {
+        return null;
+    }
+
+    /**
+     * Get request failed message setting name.
+     */
+    public function getFailedMsgSettingName(): string
+    {
+        return $this->getName() . '_failedMsg';
+    }
+
+    /**
+     * Helper to register hooks that are used in normal plugin setup.
+     */
+    private function _pluginInitialization()
+    {
+        PluginRegistry::register('importexport', new MedraExportPlugin($this), $this->getPluginPath());
+
+        Hook::add('DoiSettingsForm::setEnabledRegistrationAgencies', [$this, 'addAsRegistrationAgencyOption']);
+        Hook::add('DoiSetupSettingsForm::getObjectTypes', [$this, 'addAllowedObjectTypes']);
+        Hook::add('DoiListPanel::setConfig', [$this, 'addRegistrationAgencyName']);
+        Hook::add('Schema::get::doi', [$this, 'addToSchema']);
+        Hook::add('Doi::markRegistered', [$this, 'editMarkRegisteredParams']);
+    }
+
+    /**
+     * Get mEDRA export plugin
+     */
+    private function _getExportPlugin(): MedraExportPlugin
+    {
+        if (empty($this->_exportPlugin)) {
+            $pluginCategory = 'importexport';
+            $pluginPathName = 'MedraExportPlugin';
+            $this->_exportPlugin = PluginRegistry::getPlugin($pluginCategory, $pluginPathName);
+            // If being run from CLI, there is no context, so plugin initialization would not have been fired
+            // Note: from the release 3.4 there is no possibility to run the plugin from CLI
```

Comment:
I think this might also be the case if things are invoked from a job, but I'm not 100%.